optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateTypeTest.java

```diff
@@ -16,62 +16,81 @@
+    @After
+    public void tearDown() throws Exception {
+        PersistenceUtil.cleanUp(context);
     }
 
     protected <S extends Score, E extends AbstractTestJpaEntity<S>> Long persistAndAssert(E jpaEntity) {
-        EntityManager entityManager = entityManagerFactory.createEntityManager();
-        EntityTransaction transaction = entityManager.getTransaction();
-        transaction.begin();
-        entityManager.persist(jpaEntity);
-        transaction.commit();
+        try {
+            transactionManager.begin();
+            EntityManager em = entityManagerFactory.createEntityManager();
+            em.persist(jpaEntity);
+            transactionManager.commit();
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            throw new RuntimeException(ex.getMessage(), ex);
+        }
         Long id = jpaEntity.getId();
         assertNotNull(id);
         return id;
+
     }
 
     protected <S extends Score, E extends AbstractTestJpaEntity<S>> void findAssertAndChangeScore(
             Class<E> jpaEntityClass, Long id, S oldScore, S newScore) {
-        EntityManager entityManager = entityManagerFactory.createEntityManager();
-        EntityTransaction transaction = entityManager.getTransaction();
-        transaction.begin();
-        E jpaEntity = entityManager.find(jpaEntityClass, id);
-        assertEquals(oldScore, jpaEntity.getScore());
-        jpaEntity.setScore(newScore);
-        jpaEntity = entityManager.merge(jpaEntity);
-        transaction.commit();
+        try {
+            transactionManager.begin();
+            EntityManager em = entityManagerFactory.createEntityManager();
+            E jpaEntity = em.find(jpaEntityClass, id);
+            em.persist(jpaEntity);
+            assertEquals(oldScore, jpaEntity.getScore());
+            jpaEntity.setScore(newScore);
+            jpaEntity = em.merge(jpaEntity);
+            transactionManager.commit();
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            throw new RuntimeException(ex.getMessage(), ex);
+        }
     }
 
     protected <S extends Score, E extends AbstractTestJpaEntity<S>> void findAndAssert(
             Class<E> jpaEntityClass, Long id, S score) {
-        EntityManager entityManager = entityManagerFactory.createEntityManager();
-        EntityTransaction transaction = entityManager.getTransaction();
-        transaction.begin();
-        E jpaEntity = entityManager.find(jpaEntityClass, id);
-        assertEquals(score, jpaEntity.getScore());
-        transaction.commit();
+        try {
+            transactionManager.begin();
+            EntityManager em = entityManagerFactory.createEntityManager();
+            E jpaEntity = em.find(jpaEntityClass, id);
+            assertEquals(score, jpaEntity.getScore());
+            transactionManager.commit();
+        } catch (Exception ex) {
+            ex.printStackTrace();
+            throw new RuntimeException(ex.getMessage(), ex);
```

Comment:
printing an exception and throwing it causes it to be logged twice.
Instead of wrapping all code in try-catch-everything, can't we just add "throws TransactionException" to all methods, as this is test code?